File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.py
def solution(elements):
    answer = 0
    l = len(elements)
    elements = 2*elements
    aset = set()
    for j in range(1,l+1): 
        for i in range(l):
            aset.add(sum(elements[i:i+j]))
    return len(aset)

File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.cpp
#include <string>
#include <vector>
#include <set>
using namespace std;

int solution(vector<int> elements) {
    int answer = 0;
    set<int> sset;
    
    for (int i = 0; i < elements.size(); ++i){
        int sum = 0;
        for (int j = i; j < i+elements.size(); ++j){
            int idx = j % elements.size();
            sum += elements[idx];
            sset.insert(sum);
        }
    }
        
    return sset.size();
}